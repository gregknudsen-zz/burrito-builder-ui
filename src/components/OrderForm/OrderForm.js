import React, { Component } from 'react';

class OrderForm extends Component {
  constructor(props) {
    super();
    this.props = props;
    this.state = {
      name: '',
      ingredients: []
    };
  }

  handleNameChange = e => {
    e.preventDefault();
    const name = e.target.value;
    console.log({name});
    this.setState({name});
  }

  handleIngredientChange = e => {
    e.preventDefault();
    const ingredient = e.target.name;
    const addedIngredients = this.state.ingredients || [];
    addedIngredients.push(ingredient);
    console.log(addedIngredients);
    this.setState({ingredients: addedIngredients})
  }

  // const onChange = evt => {
  //   console.log(`TASK 8- Obtain 'name' and 'value' from the target of the event,
  //     and utilize the correct callback to update the state of the form.`)
  //     const {name, value} = evt.target
  //     setValues({...values, [name]: value});
  // }


  handleSubmit = e => {
    e.preventDefault();
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({name: '', ingredients: []});
  }

  render() {
    const possibleIngredients = ['beans', 'steak', 'carnitas', 'sofritas', 'lettuce', 'queso fresco', 'pico de gallo', 'hot sauce', 'guacamole', 'jalapenos', 'cilantro', 'sour cream'];
    const ingredientButtons = possibleIngredients.map(ingredient => {
      return (
        <button key={ingredient} name={ingredient} onClick={e => this.handleIngredientChange(e)}>
          {ingredient}
        </button>
      )
    });

    return (
      <form>
        <input
          type='text'
          placeholder='Name'
          name='name'
          value={this.state.name}
          onChange={e => this.handleNameChange(e)}
        />

        { ingredientButtons }

        <p>Order: { this.state.ingredients.join(', ') || 'Nothing selected' }</p>

        <button onClick={e => this.handleSubmit(e)}>
          Submit Order
        </button>
      </form>
    )
  }
}

export default OrderForm;
