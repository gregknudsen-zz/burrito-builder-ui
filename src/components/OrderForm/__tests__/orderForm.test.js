import React from 'react';
import { render, cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import OrderForm from '../OrderForm.js';

describe('OrderForm Component', () => {
  it('renders', () => {
    render(<OrderForm />)
  })
  it('matches snapshot DOM node structure', () => {
    // Arrange the snapshot - declare variables
    const { asFragment } = render(<OrderForm />);
    // Assert the match
      expect(asFragment()).toMatchSnapshot();
  })
  it.skip('successfully adds an order when both fields have values', () => {
    const newOrder = {
      name: 'Bob',
      ingredients: ['mushrooms']
    }
    
  })


});