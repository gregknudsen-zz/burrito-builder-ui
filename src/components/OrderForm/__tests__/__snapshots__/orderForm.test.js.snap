// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderForm Component matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <form>
    <input
      name="name"
      placeholder="Name"
      type="text"
      value=""
    />
    <button
      name="beans"
    >
      beans
    </button>
    <button
      name="steak"
    >
      steak
    </button>
    <button
      name="carnitas"
    >
      carnitas
    </button>
    <button
      name="sofritas"
    >
      sofritas
    </button>
    <button
      name="lettuce"
    >
      lettuce
    </button>
    <button
      name="queso fresco"
    >
      queso fresco
    </button>
    <button
      name="pico de gallo"
    >
      pico de gallo
    </button>
    <button
      name="hot sauce"
    >
      hot sauce
    </button>
    <button
      name="guacamole"
    >
      guacamole
    </button>
    <button
      name="jalapenos"
    >
      jalapenos
    </button>
    <button
      name="cilantro"
    >
      cilantro
    </button>
    <button
      name="sour cream"
    >
      sour cream
    </button>
    <p>
      Order: Nothing selected
    </p>
    <input
      type="submit"
      value="Submit"
    />
  </form>
</DocumentFragment>
`;
